### PR TITLE
Custom Build refinements

### PR DESCRIPTION
**Pull request by Palmer-JC**
- Changed if test to use getClassName() instead of instanceof
- Improved the directions of the page
- Added a output directory text field on page

### CODE REVIEW

File: src/babylon.engine.ts [hunk 1]
```diff
@@ -3362,26 +3359,32 @@
 
         // Loading screen
         public displayLoadingUI(): void {
+            if (!this._loadingScreen) this._loadingScreen = new DefaultLoadingScreen(this._renderingCanvas)
```
Comment (deltakosh) [inline src/babylon.engine.ts]: what about creating a loadingScreen property? this way you could instantiate it on first get?

File: src/Mesh/babylon.subMesh.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@
         public getMaterial(): Material {
             var rootMaterial = this._renderingMesh.material;
 
-            if (rootMaterial && rootMaterial instanceof MultiMaterial) {
+            if (rootMaterial && rootMaterial.getClassName() === "MultiMaterial") {
```
Comment (deltakosh) [inline src/Mesh/babylon.subMesh.ts]: for performance reason we should just test for getSubMaterial existence

File: src/Materials/babylon.materialHelper.ts [hunk 3]
```diff
@@ -111,11 +111,11 @@
                     defines["DIRLIGHT" + lightIndex] = false;
 
                     var type;
-                    if (light instanceof SpotLight) {
+                    if (light.getClassName() === "SpotLight") {
```
Comment (deltakosh) [inline src/Materials/babylon.materialHelper.ts]: just wondering if string testing is a good idea regarding perd. Perhaps we could introduce a function  to return the type directly

File: src/Materials/babylon.materialHelper.ts [hunk 4]
```diff
@@ -305,16 +305,16 @@
         }
 
         public static BindLightProperties(light: Light, effect: Effect, lightIndex: number): void {
-            if (light instanceof PointLight) {
+            if (light.getClassName() === "PointLight") {
```
Comment (deltakosh) [inline src/Materials/babylon.materialHelper.ts]: we also should do it at light level to avoid string testing

File: src/babylon.engine.ts (hunk 1 above)
Comment (Palmer-JC) [inline src/babylon.engine.ts]: Actually there already was one.  Done in next PR

File: src/Mesh/babylon.subMesh.ts (hunk 2 above)
Comment (Palmer-JC) [inline src/Mesh/babylon.subMesh.ts]: Had to cast to MultiMaterial to get getSubMaterial existence test to work, but casting is just typescript

Comment (deltakosh) [inline src/Mesh/babylon.subMesh.ts]: yep so no perf issues

File: src/Materials/babylon.materialHelper.ts (hunk 3 above)
Comment (Palmer-JC) [inline src/Materials/babylon.materialHelper.ts]: for light tests, I now use getTypeID instead.  It already exists & is an integer.  Think I need to close this PR and open a new one to add another commit.

Comment (deltakosh) [inline src/Materials/babylon.materialHelper.ts]: Ok let me merge this one then

**Comment by deltakosh** [comment]
I've just did the change in subMesh as I was working on the file already

**Pull request merged**